src/SignalR/common/Http.Connections/test/HttpConnectionDispatcherTests.cs

```diff
@@ -2642,6 +2657,495 @@ public async Task LongRunningActivityTagSetOnExecuteAsync()
+
+            var negotiateResponse = NegotiateProtocol.ParseResponse(stream.ToArray());
+
+            Assert.True(manager.TryGetConnection(negotiateResponse.ConnectionToken, out var context));
+
+            Assert.Null(context.AuthenticationExpiration);
+
+            await connection.DisposeAsync();
+        }
+
+        private class GetNegotiateHttpHandler : DelegatingHandler
+        {
+            private readonly MemoryStream _stream;
+            private bool _read;
+
+            public GetNegotiateHttpHandler(HttpMessageHandler handler, MemoryStream stream)
+                : base(handler)
+            {
+                _stream = stream;
+            }
+
+            protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                var response = await base.SendAsync(request, cancellationToken);
+                if (!_read)
+                {
+                    await response.Content.CopyToAsync(_stream);
+                    response.Content = new ByteArrayContent(_stream.ToArray());
+                    _stream.Position = 0;
+                    _read = true;
+                }
+                return response;
+            }
+        }
+
+        private class ForwardingLoggerProvider : ILoggerProvider
+        {
+            private readonly ILoggerFactory _loggerFactory;
+
+            public ForwardingLoggerProvider(ILoggerFactory loggerFactory)
+            {
+                _loggerFactory = loggerFactory;
+            }
+
+            public void Dispose()
+            {
+            }
+
+            public ILogger CreateLogger(string categoryName)
+            {
+                return _loggerFactory.CreateLogger(categoryName);
+            }
+        }
+
+        private static IHost CreateHost(Action<IServiceCollection> configureServices, Action<IEndpointRouteBuilder> configureEndpoints,
+            ILoggerFactory loggerFactory)
+        {
+            return new HostBuilder()
+                .ConfigureWebHost(webHostBuilder =>
+                {
+                    webHostBuilder
+                    .UseKestrel()
+                    .ConfigureLogging(o =>
+                    {
+                        o.AddProvider(new ForwardingLoggerProvider(loggerFactory));
+                    })
+                    .ConfigureServices(services =>
+                    {
+                        services.AddConnections();
+                        configureServices(services);
+                        services.AddAuthorization();
+
+                        // Since tests run in parallel, it's possible multiple servers will startup and read files being written by another test
+                        // Use a unique directory per server to avoid this collision
+                        services.AddDataProtection()
+                            .PersistKeysToFileSystem(Directory.CreateDirectory(Path.GetRandomFileName()));
```

Comment:
Maybe slightly cleaner to just use  

`services.AddSingleton<IDataProtectionProvider, EphemeralDataProtectionProvider>();`

like here: https://github.com/dotnet/aspnetcore/blob/c6713591462b6e547860f3f289472865af3602a0/src/Components/test/testassets/TestServer/ServerStartup.cs#L30